base/utferror.jl

```diff
@@ -0,0 +1,30 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+##\brief      Error messages for Unicode / UTF support
```

Comment:
Umm... I've got ask, and this may sound (and be!) utterly snarky, but *what* documentation?
Almost zero documentation is not better than having at least some documentation in doxygen format.
(I'm talking here about *internal* documentation of structures and functions in Base, not general
documentation, or documentation in packages, such as @jiahao's beautiful documentation in factorization.jl [his is the only real good documentation I've seen so far in the code... but admittedly I haven't been looking at packages hardly at all, mostly Base]).

Actually, I'd like to propose letting my doxygen markup go in as is for now, but as soon as some standard is decided on [and I like parsing markdown idea in the last comments in #8966 by @one-more-minute, as long as the format is actually agreed on, and a parser to extract the metadata written], I will immediately change it.